sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -316,6 +317,21 @@ trait Row extends Serializable {
   }
 
   /**
+   * Returns the Optional value at position i.
+   * If the value is null or if it cannot be cast to the requested type, None is returned.
+   */
+  def getAsOpt[T](i: Int): Option[T] = {
+    val value = {
+      if (i < length) {
+        if (isNullAt(i)) None
+        else catching(classOf[ClassCastException]) opt getAs[T](i)
```

Comment:
We typically avoid infix notation and this style of `if`.

https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide

Also, why do we want to hide `ClassCastException` from the user, this seems more confusing than helpful.